softlayer/vm/softlayer_hardware_creator.go

```diff
@@ -79,8 +101,73 @@ func (c *baremetalCreator) Create(agentID string, stemcell bslcstem.Stemcell, cl
 	return hardware, nil
 }
 
+func (c *baremetalCreator) createByOSReload(agentID string, stemcell bslcstem.Stemcell, cloudProps VMCloudProperties, networks Networks, env Environment) (VM, error) {
+	if len(cloudProps.BaremetalStemcell) == 0 {
+		return nil, bosherr.Error("No stemcell provided to do os_reload.")
+	}
+
+	hardwareService, err := c.softLayerClient.GetSoftLayer_Hardware_Service()
+	if err != nil {
+		return nil, bosherr.WrapError(err, "Creating HardwareService from SoftLayer client")
+	}
+
+	hardware, err := hardwareService.FindByIpAddress(networks.First().IP)
+	if err != nil || hardware.Id == 0 {
+		return nil, bosherr.WrapErrorf(err, "Could not find hardware by ip address: %s", networks.First().IP)
+	}
+
+	c.logger.Info(SOFTLAYER_VM_CREATOR_LOG_TAG, fmt.Sprintf("OS reload on Hardware %d using stemcell %d", hardware.Id, stemcell.ID()))
+
+	vm, found, err := c.vmFinder.Find(hardware.Id)
+	if err != nil || !found {
+		return nil, bosherr.WrapErrorf(err, "Cannot find hardware with id: %d", hardware.Id)
+	}
+
+	err = vm.ReloadOSForBaremetal(cloudProps.BaremetalStemcell, cloudProps.BaremetalNetbootImage)
+	if err != nil {
+		return nil, bosherr.WrapError(err, "Failed to reload OS")
+	}
+
+	vm, found, err = c.vmFinder.Find(hardware.Id)
+	if err != nil || !found {
+		return nil, bosherr.WrapErrorf(err, "Cannot find hardware with id: %d.", vm.ID())
+	}
+
+	// Update mbus url setting
+	mbus, err := ParseMbusURL(c.agentOptions.Mbus, cloudProps.BoshIp)
+	if err != nil {
+		return nil, bosherr.WrapErrorf(err, "Cannot construct mbus url.")
+	}
+	c.agentOptions.Mbus = mbus
+	// Update blobstore setting
+	switch c.agentOptions.Blobstore.Provider {
+	case BlobstoreTypeDav:
+		davConf := DavConfig(c.agentOptions.Blobstore.Options)
+		UpdateDavConfig(&davConf, cloudProps.BoshIp)
+	}
+
+	agentEnv := CreateAgentUserData(agentID, cloudProps, networks, env, c.agentOptions)
+	if err != nil {
```

Comment:
This code is wrong, the test for err is not from previous statement which does not return any error: 
```
agentEnv := CreateAgentUserData(agentID, cloudProps, networks, env, c.agentOptions)
```